thresholding.lisp

```diff
@@ -68,7 +68,9 @@ The elements at indexex 0 and 255 are set to +j-fn-max-value+."
            (incf apriori-1 (aref histogram idx))
            (if (zerop apriori-1)
                (setf (aref samples idx) +j-fn-max-value+)
-               (let* ((apriori-2 (- 1d0 apriori-1))
+               (let* ((apriori-2 (if (= apriori-1 1.0d0)
+				     least-positive-double-float
+				     (- 1d0 apriori-1)))
```

Comment:
@slyrus If `apriori-1` is equal to 1d0, then `apriori-2` is going to be equal to 0d0. This means that computing `mean-2` like this:

```lisp
                      (mean-2 (/ (loop for i from (1+ idx) to (- histlen 1)
                                    summing (* (aref histogram i) i))
                                 apriori-2))
```

is going to divide by zero. This patch avoids it by instead nudging that zero upwards into a machine epsilon.